bentoml/cli/utils.py

```diff
@@ -92,6 +93,43 @@ def parse_key_value_pairs(key_value_pairs_str):
     return result
 
 
+def _echo_docker_api_result(docker_generator):
+    layers = {}
+    for line in docker_generator:
+        if "stream" in line:
+            cleaned = line['stream'].rstrip("\n")
+            if cleaned != "":
+                _echo(cleaned)
+        if "status" in line and line["status"] == "Pushing":
+            progress = line["progressDetail"]
+            layers[line["id"]] = progress["current"], progress["total"]
+            cur, total = zip(*layers.values())
+            cur, total = (
+                humanfriendly.format_size(sum(cur)),
+                humanfriendly.format_size(sum(total)),
+            )
+            _echo(f"Pushed {cur} / {total}")
+        if "errorDetail" in line:
+            error = line["errorDetail"]
+            raise BentoMLException(f'Could not push Docker image: {error["message"]}')
```

Comment:
Is there a similar error for building the docker image?

Reply:
`.build` seems to already raise a `docker.errors.APIError` but `.push` will instead stream `"errorDetail"`, so we should catch and raise an error here.